src/main/java/org/myrobotlab/encoder/TimerEncoder.java

```diff
@@ -0,0 +1,24 @@
+package org.myrobotlab.encoder;
+
+import org.myrobotlab.framework.interfaces.Attachable;
+
+public class TimerEncoder extends GenericEncoder {
+  
+  transient Thread thread;
+
+  // thread - executor ?
+  
+  // resolution
+  
+  // start - stop
+  
+  // move from x to y at velocity z
+  
+  // use acceleration - slope - max velocity
+  
+  public void attach(Attachable attachable) throws Exception {
+    
```

Comment:
it will .. I have to finish it in order to provide the a transportable "waitForPos()"  for all controllers  moveToBlocking will depend on it ... ya needs code :)